src/pages/ExamsPage/components/AddAllowanceModal.jsx

```diff
@@ -0,0 +1,239 @@
+            </Form.Group>
+          ) : null }
+          <Form.Group controlId="form-allowance-type">
+            <Form.Label>{ formatMessage(messages.addAllowanceAllowanceTypeField) }</Form.Label>
+            <Form.Control
+              as="select"
+              name="allowance-type"
+              onChange={handleChange}
+              defaultValue="additional-minutes"
+              data-testid="allowance-type"
+            >
+              <option value="additional-minutes">{ formatMessage(messages.addAllowanceAdditionalMinutesOption) }</option>
+              <option value="time-multiplier">{ formatMessage(messages.addAllowanceTimeMultiplierOption) }</option>
+            </Form.Control>
+          </Form.Group>
+          { form['allowance-type'] === 'additional-minutes'
+            ? (
+              <Form.Group controlId="form-allowance-value-minutes" isInvalid={additionalTimeError}>
+                <Form.Label>{ formatMessage(messages.addAllowanceMinutesField) }</Form.Label>
+                <Form.Control
+                  name="additional-time-minutes"
+                  defaultValue=""
+                  onChange={handleChange}
+                  data-testid="additional-time-minutes"
+                />
+                { additionalTimeError && <Form.Control.Feedback type="invalid">{ formatMessage(messages.addAllowanceMinutesErrorFeedback) }</Form.Control.Feedback> }
+              </Form.Group>
+            )
+            : (
+              <Form.Group controlId="form-allowance-value-multiplier" isInvalid={additionalTimeError}>
+                <Form.Label>{ formatMessage(messages.addAllowanceMultiplierField) }</Form.Label>
+                <Form.Control name="additional-time-multiplier" defaultValue="" onChange={handleChange} />
+                {
+                  additionalTimeError
+                    ? (
+                      <Form.Control.Feedback type="invalid">
+                        { formatMessage(messages.addAllowanceMultiplierFeedback) }
+                      </Form.Control.Feedback>
+                    )
+                    : (
+                      <Form.Control.Feedback>
+                        { formatMessage(messages.addAllowanceMultiplierErrorFeedback) }
+                      </Form.Control.Feedback>
+                    )
+                }
+              </Form.Group>
+            )}
+        </Form>
+      </ModalDialog.Body>
+
+      <ModalDialog.Footer>
+        <ActionRow>
+          <ModalDialog.CloseButton variant="tertiary">
+            { formatMessage(messages.addAllowanceCloseButton) }
+          </ModalDialog.CloseButton>
+          <StatefulButton
+            data-testid="create-allowance-stateful-button"
+            state={createAllowanceRequestStatus()}
+            {...addAllowanceButtonProps}
+            variant="primary"
+            onClick={onSubmit}
```

Comment:
If we process the form in the button click I think we might have an issue, since hitting enter on an input would submit the `<Form>` used above. And if not, the user might expect this behavior.

I would consider changing the form above as `<Form id="add-allowance-form" onSubmit={onSubmit}>` and then use the button like this.

Also, if Paragon's `<StatefulButton>` passes on props internally to a nromal button, you could also use the props `type="submit"` and `form="add-allowance-form"` to let the form handle the submit.

Reply:
Hitting enter on the form input does not cause the form to submit for the current code. 

I can change the code based on your suggestion, but I would need to add `e.preventDefault()` to `onSubmit`, otherwise the entire page refreshes on form submission.

Reply:
That's correct. I forgot the preventDefault on the onSubmit().

Reply:
preventDefault also prevents hitting enter in a field to submit the form, though, so I think I'm going to opt to keep the button functionality as is. The form data is still being submitted as expected.

Reply:
I think preventDefault has to be used within the onSubmit in the form, which was already called by hitting enter on the control.